tests/test_association_task.py

```diff
@@ -286,66 +396,68 @@ def test_update_dia_objects(self):
-            dia_source["psFlux"] = 20000
-            dia_source["psFluxErr"] = 100
+            dia_source["snr"] = 10
+            dia_source["midPointTai"] = \
+                self.exposure.getInfo().getVisitInfo().getDate().get(system=dafBase.DateTime.MJD)
+            dia_source["x"] = 0
+            dia_source["y"] = 0
+            dia_source["psFlux"] = 20000 / self.flux0
+            dia_source["psFluxErr"] = \
+                np.sqrt((100 / self.flux0) ** 2 + (20000 * self.flux0_err / self.flux0 ** 2) ** 2)
+            dia_source["apFlux"] = 20000 / self.flux0
+            dia_source["apFluxErr"] = \
+                np.sqrt((100 / self.flux0) ** 2 + (20000 * self.flux0_err / self.flux0 ** 2) ** 2)
             dia_source["filterName"] = "g"
             dia_source["filterId"] = 1
 
         # Store them in the DB containing pre-existing objects.
-        assoc_db_config = AssociationDBSqliteConfig()
-        assoc_db_config.db_name = self.db_file
-        assoc_db_config.filter_names = self.filter_names
-        assoc_db = AssociationDBSqliteTask(config=assoc_db_config)
-        assoc_db.create_tables()
+        ppdb = Ppdb(config=self.ppdbConfig,
+                    afw_schemas=dict(DiaObject=make_dia_object_schema(),
+                                     DiaSource=make_dia_source_schema()))
         # Load the Existing DIAObjects for use in the update method.
-        loaded_dia_objects = assoc_db.load_dia_objects(self.exposure)
+        loaded_dia_objects = ppdb.getDiaObjects(self.index_ranges)
 
         # Store the new DIASources with associations and an exposure.
-        assoc_db.store_dia_sources(dia_sources,
-                                   [1, 2, 3, 4, 14],
-                                   self.exposure)
-        assoc_db.close()
-        del assoc_db
+        ppdb.storeDiaSources(dia_sources)
 
         # Create our task and update the stored DIAObjects.
         assoc_config = AssociationConfig()
-        assoc_config.level1_db.value.db_name = self.db_file
-        assoc_config.level1_db.filter_names = self.filter_names
         assoc_task = AssociationTask(config=assoc_config)
         assoc_task.update_dia_objects(loaded_dia_objects,
                                       [1, 2, 3, 4, 14],
-                                      self.exposure)
+                                      self.exposure,
+                                      ppdb)
 
         # Retrieve the DIAObjects from the DB.
-        output_dia_objects = assoc_task.level1_db.load_dia_objects(self.exposure)
+        output_dia_objects = ppdb.getDiaObjects(self.index_ranges)
 
         # Data and column names to test.
         test_column_names = [
-            'id', 'nDiaSources',
-            'psFluxMean_g', 'psFluxMeanErr_g', 'psFluxSigma_g',
-            'psFluxMean_r', 'psFluxMeanErr_r', 'psFluxSigma_r']
+            'id', 'gPSFluxMean', 'gPSFluxMeanErr', 'gPSFluxSigma',
+            'rPSFluxMean', 'rPSFluxMeanErr', 'rPSFluxSigma']
```

Comment:
Does this depend on the content of the schema files or some other external source? I'm worried about it getting out of sync.

Reply:
Actually, why not just use `test_dia_object_values[0].keys()`? You'd have one fewer copy to keep up to date.

Reply:
Sounds good.